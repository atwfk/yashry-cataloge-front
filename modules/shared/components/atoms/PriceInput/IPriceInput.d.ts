declare namespace IPriceInput {
  export interface IProps {
    minPrice: number;
    maxPrice: number;
    from: number;
    to: number;
    updateMinPrice: (minPrice: string) => void;
    updateMaxPrice: (maxPrice: string) => void;
  }
}

export { IPriceInput };
