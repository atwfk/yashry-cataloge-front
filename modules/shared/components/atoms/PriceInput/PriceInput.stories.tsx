import type { ReactElement } from "react";
import type { Story } from "@storybook/react";
import PriceInput from "./PriceInput";
import { IPriceInput } from "./IPriceInput";

const args = {
  title: "Atoms/PriceInput",
  component: PriceInput,
};

export default args;

const Template: Story<IPriceInput.IProps> = (args): ReactElement => (
  <PriceInput {...args} />
);

export const priceInput = Template.bind({});
priceInput.args = {
  minPrice: 60,
  maxPrice: 150,
  from: 90,
  to: 120,
};
