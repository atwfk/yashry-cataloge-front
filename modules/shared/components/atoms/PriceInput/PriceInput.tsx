import React, { useState } from "react";
import type { FC, ReactElement, FocusEvent, ChangeEvent } from "react";
import { IPriceInput } from "./IPriceInput";
import P from "../Paragraph/Paragraph";

const PriceInput: FC<IPriceInput.IProps> = ({
  minPrice,
  maxPrice,
  from,
  to,
  updateMinPrice,
  updateMaxPrice,
}): ReactElement => {
  const [minPriceVal, setMinPriceVal] = useState(from);
  const [maxPriceVal, setMamPriceVal] = useState(to);

  return (
    <div className="flex flex-col">
      <P fontColor="dark" fontSize="md" fontWeight="medium" classes="mb-2">
        Prices: ${minPrice} - ${maxPrice}
      </P>
      <div>
        <input
          type="number"
          name="min-price"
          id="minPrice"
          value={minPriceVal}
          className="border-2 border-dark-grey rounded-md w-16 mr-2"
          onBlur={(e: FocusEvent<HTMLInputElement>): void => {
            updateMinPrice(e.currentTarget.value);
          }}
          onChange={(e: ChangeEvent<HTMLInputElement>): void => {
            setMinPriceVal(+e.currentTarget.value);
          }}
        />
        <input
          type="number"
          name="max-price"
          id="maxPrice"
          value={maxPriceVal}
          className="border-2 border-dark-grey rounded-md w-16"
          onBlur={(e: FocusEvent<HTMLInputElement>) => {
            updateMaxPrice(e.currentTarget.value);
          }}
          onChange={(e: ChangeEvent<HTMLInputElement>): void => {
            setMamPriceVal(+e.currentTarget.value);
          }}
        />
      </div>
    </div>
  );
};

export default PriceInput;
