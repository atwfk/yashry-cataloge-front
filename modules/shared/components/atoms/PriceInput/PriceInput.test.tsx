import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import PriceInput from "./PriceInput";

describe("PriceInput snapshots", () => {
  it("should redner PriceInput with correct price values", () => {
    const tree = renderer
      .create(
        <PriceInput
          minPrice={6}
          maxPrice={1000}
          from={50}
          to={150}
          updateMaxPrice={(): boolean => true}
          updateMinPrice={(): boolean => true}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("PriceInput RTL", () => {
  it("should not accept strings when typing in price inputs", () => {
    render(
      <PriceInput
        minPrice={6}
        maxPrice={1000}
        from={50}
        to={150}
        updateMaxPrice={(): boolean => true}
        updateMinPrice={(): boolean => true}
      />,
    );

    const minPriceInput = screen.getByTestId("min-price");
    const maxPriceInput = screen.getByTestId("max-price");

    userEvent.type(minPriceInput, "string");
    userEvent.type(maxPriceInput, "string");

    expect(minPriceInput).toHaveValue(50);
    expect(maxPriceInput).toHaveValue(150);
  });

  it("should accept numbers when typing in price inputs", () => {
    render(
      <PriceInput
        minPrice={6}
        maxPrice={1000}
        from={50}
        to={150}
        updateMaxPrice={(): boolean => true}
        updateMinPrice={(): boolean => true}
      />,
    );

    const minPriceInput = screen.getByTestId("min-price");
    const maxPriceInput = screen.getByTestId("max-price");

    userEvent.clear(minPriceInput);
    userEvent.clear(maxPriceInput);
    userEvent.type(minPriceInput, "300");
    userEvent.type(maxPriceInput, "400");

    expect(minPriceInput).toHaveValue(300);
    expect(maxPriceInput).toHaveValue(400);
  });

  it("should call updateMaxPrice once and updateMinPrice once", () => {
    const updateMinPrice = jest.fn();
    const updateMaxPrice = jest.fn();

    render(
      <PriceInput
        minPrice={6}
        maxPrice={1000}
        from={50}
        to={150}
        updateMaxPrice={updateMaxPrice}
        updateMinPrice={updateMinPrice}
      />,
    );

    const minPriceInput = screen.getByTestId("min-price");
    const maxPriceInput = screen.getByTestId("max-price");

    userEvent.click(minPriceInput);
    userEvent.tab();
    userEvent.click(maxPriceInput);
    userEvent.tab();

    expect(updateMinPrice).toHaveBeenCalledTimes(1);
    expect(updateMaxPrice).toHaveBeenCalledTimes(1);
  });
});
