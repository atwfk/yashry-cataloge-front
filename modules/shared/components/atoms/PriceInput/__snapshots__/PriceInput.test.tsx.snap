// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceInput snapshots should redner PriceInput with correct price values 1`] = `
<div
  className="flex flex-col"
>
  <p
    className="text-base text-dark font-medium mb-2"
  >
    Prices: $
    6
     - $
    1000
  </p>
  <div>
    <input
      className="border-2 border-dark-grey rounded-md w-16 mr-2"
      data-testid="min-price"
      id="minPrice"
      name="min-price"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
      value={50}
    />
    <input
      className="border-2 border-dark-grey rounded-md w-16"
      data-testid="max-price"
      id="maxPrice"
      name="max-price"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
      value={150}
    />
  </div>
</div>
`;
